### PR TITLE
Ensure that macOS downstream builds do not hardcode a specific minor.patch version when linking to the library

### LINKED ISSUE #18: macOS builds should have run_exports max_pin="x.x.x"
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

I do not know if this is a problem upstream or it is a deliberate decision, but some some reason it seems that macOS CMake config files for `sdl2_ttf` ask downstream build systems to link the specific patch version of the library, creating a runtime dependency not on any sdl2_ttf version newer then the version used, but exactly on the patch version of sdl2_ttf used to build the package. 

To understand why is happening, see the content of `lib\cmake\SDL2_ttf\SDL2-shared-targets-release.cmake` file from `osx-64/sdl2_ttf-2.22.0-hcf92afd_0.conda`, its content is:
~~~
#----------------------------------------------------------------
# Generated CMake target import file for configuration "Release".
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Import target "SDL2_ttf::SDL2_ttf" for configuration "Release"
set_property(TARGET SDL2_ttf::SDL2_ttf APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(SDL2_ttf::SDL2_ttf PROPERTIES
  IMPORTED_LINK_DEPENDENT_LIBRARIES_RELEASE "SDL2::SDL2"
  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libSDL2_ttf-2.0.2201.0.0.dylib"
  IMPORTED_SONAME_RELEASE "@rpath/libSDL2_ttf-2.0.2201.0.0.dylib"
  )

list(APPEND _cmake_import_check_targets SDL2_ttf::SDL2_ttf )
list(APPEND _cmake_import_check_files_for_SDL2_ttf::SDL2_ttf "${_IMPORT_PREFIX}/lib/libSDL2_ttf-2.0.2201.0.0.dylib" )

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
~~~

Note how the `IMPORTED_SONAME_RELEASE` refers to the exact name of the library, not to the `libSDL2_ttf-2.0.dylib` or `libSDL2_ttf.dylib` symlink. On the other hand, the same `lib\cmake\SDL2_ttf\SDL2-shared-targets-release.cmake` file in the `linux-64/sdl2_ttf-2.22.0-ha14f88b_0.conda` linux package is :

~~~
#----------------------------------------------------------------
# Generated CMake target import file for configuration "Release".
#----------------------------------------------------------------

# Commands may need to know the format version.
set(CMAKE_IMPORT_FILE_VERSION 1)

# Import target "SDL2_ttf::SDL2_ttf" for configuration "Release"
set_property(TARGET SDL2_ttf::SDL2_ttf APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
set_target_properties(SDL2_ttf::SDL2_ttf PROPERTIES
  IMPORTED_LINK_DEPENDENT_LIBRARIES_RELEASE "SDL2::SDL2"
  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libSDL2_ttf-2.0.so.0.2200.0"
  IMPORTED_SONAME_RELEASE "libSDL2_ttf-2.0.so.0"
  )

list(APPEND _cmake_import_check_targets SDL2_ttf::SDL2_ttf )
list(APPEND _cmake_import_check_files_for_SDL2_ttf::SDL2_ttf "${_IMPORT_PREFIX}/lib/libSDL2_ttf-2.0.so.0.2200.0" )

# Commands beyond this point should not need to know the version.
set(CMAKE_IMPORT_FILE_VERSION)
~~~

In this case, the `IMPORTED_SONAME_RELEASE` referece to the generic x.x name of the library, not to the x.x.x name of the library.

### Installed packages

```shell
osx-64/sdl2_ttf-2.22.0-hcf92afd_0.conda
```

### Environment info

```shell
Not relevant.
```

### PR DESCRIPTION
**Pull request by traversaro**
Fix https://github.com/conda-forge/sdl2_ttf-feedstock/issues/18 by backporting https://github.com/libsdl-org/SDL_ttf/pull/349 and adding symlinks for existing builds.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
Ok, I validated the solution on osx-arm64, I reproduced the problem by running:

~~~
(pygame) icub@iCubs-Mac-mini ~ % python -c "import pygame;pygame.font.init()"
pygame 2.5.2 (SDL 2.28.5, Python 3.12.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
<string>:1: RuntimeWarning: use font: dlopen(/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so, 0x0002): Library not loaded: '@rpath/libSDL2_ttf-2.0.15.0.0.dylib'
  Referenced from: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so'
  Reason: tried: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/local/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file)
(ImportError: dlopen(/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so, 0x0002): Library not loaded: '@rpath/libSDL2_ttf-2.0.15.0.0.dylib'
  Referenced from: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so'
  Reason: tried: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/local/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file))
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/__init__.py", line 70, in __getattr__
    raise NotImplementedError(missing_msg)
NotImplementedError: font module not available (ImportError: dlopen(/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so, 0x0002): Library not loaded: '@rpath/libSDL2_ttf-2.0.15.0.0.dylib'
  Referenced from: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/font.cpython-312-darwin.so'
  Reason: tried: '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/lib/python3.12/site-packages/pygame/../../../libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/Users/icub/miniforge3/envs/pygame/bin/../lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/local/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file), '/usr/lib/libSDL2_ttf-2.0.15.0.0.dylib' (no such file))
~~~

In this env:


~~~
(pygame) icub@iCubs-Mac-mini ~ % conda info

     active environment : pygame
    active env location : /Users/icub/miniforge3/envs/pygame
            shell level : 1
       user config file : /Users/icub/.condarc
 populated config files : /Users/icub/miniforge3/.condarc
                          /Users/icub/.condarc
          conda version : 24.1.2
    conda-build version : not installed
         python version : 3.10.14.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=m1
                          __conda=24.1.2=0
                          __osx=12.6=0
                          __unix=0=0
       base environment : /Users/icub/miniforge3  (writable)
      conda av data dir : /Users/icub/miniforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/icub/miniforge3/pkgs
                          /Users/icub/.conda/pkgs
       envs directories : /Users/icub/miniforge3/envs
                          /Users/icub/.conda/envs
               platform : osx-arm64
             user-agent : conda/24.1.2 requests/2.31.0 CPython/3.10.14 Darwin/21.6.0 OSX/12.6 solver/libmamba conda-libmamba-solver/24.1.0 libmambapy/1.5.7
                UID:GID : 501:20
             netrc file : None
           offline mode : False


(pygame) icub@iCubs-Mac-mini ~ % conda list
# packages in environment at /Users/icub/miniforge3/envs/pygame:
#
# Name                    Version                   Build  Channel
aom                       3.8.2                h078ce10_0    conda-forge
bzip2                     1.0.8                h93a5062_5    conda-forge
ca-certificates           2024.2.2             hf0a4a13_0    conda-forge
cairo                     1.18.0               hd1e100b_0    conda-forge
dav1d                     1.2.1                hb547adb_0    conda-forge
expat                     2.6.2                hebf3989_0    conda-forge
fluidsynth                2.3.4                h9a29e92_1    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 h77eed37_1    conda-forge
fontconfig                2.14.2               h82840c6_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
freetype                  2.12.1               hadb7bae_2    conda-forge
gettext                   0.21.1               h0186832_0    conda-forge
graphite2                 1.3.13            h9f76cd9_1001    conda-forge
harfbuzz                  8.3.0                h8f0ba13_0    conda-forge
icu                       73.2                 hc8870d7_0    conda-forge
lame                      3.100             h1a8c8d9_1003    conda-forge
lerc                      4.0.0                h9a09cb3_0    conda-forge
libavif16                 1.0.4                hff135a0_2    conda-forge
libcxx                    16.0.6               h4653b0c_0    conda-forge
libdeflate                1.20                 h93a5062_0    conda-forge
libexpat                  2.6.2                hebf3989_0    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libflac                   1.4.3                hb765f3a_0    conda-forge
libglib                   2.80.0               hfc324ee_1    conda-forge
libiconv                  1.17                 h0d3ecfb_2    conda-forge
libjpeg-turbo             3.0.0                hb547adb_1    conda-forge
libmad                    0.15.1b           h1a8c8d9_1001    conda-forge
libogg                    1.3.4                h27ca646_1    conda-forge
libopus                   1.3.1                h27ca646_1    conda-forge
libpng                    1.6.43               h091b4b1_0    conda-forge
libsndfile                1.2.2                h9739721_1    conda-forge
libsqlite                 3.45.2               h091b4b1_0    conda-forge
libtiff                   4.6.0                h07db509_3    conda-forge
libvorbis                 1.3.7                h9f76cd9_0    conda-forge
libwebp-base              1.3.2                hb547adb_0    conda-forge
libzlib                   1.2.13               h53f4e23_5    conda-forge
mpg123                    1.32.4               hebf3989_0    conda-forge
ncurses                   6.4.20240210         h078ce10_0    conda-forge
openssl                   3.2.1                h0d3ecfb_1    conda-forge
opusfile                  0.12                 h5643135_2    conda-forge
pcre2                     10.43                h26f9a81_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pixman                    0.43.4               hebf3989_0    conda-forge
portaudio                 19.6.0               h13dd4ca_9    conda-forge
portmidi                  2.0.4                h13dd4ca_2    conda-forge
pygame                    2.5.2           py312h26bdd92_2    conda-forge
python                    3.12.2          hdf0ec26_0_cpython    conda-forge
python_abi                3.12                    4_cp312    conda-forge
rav1e                     0.6.6                h69fbcac_2    conda-forge
readline                  8.2                  h92ec313_1    conda-forge
sdl2                      2.28.5               hebf3989_1    conda-forge
sdl2_image                2.8.0                h5a3017b_0    conda-forge
sdl2_mixer                2.6.3                h4fe3bdc_1    conda-forge
sdl2_ttf                  2.22.0               h4df98e8_0    conda-forge
setuptools                69.2.0             pyhd8ed1ab_0    conda-forge
svt-av1                   2.0.0                h078ce10_0    conda-forge
tk                        8.6.13               h5083fa2_1    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h57fd34a_0    conda-forge
zlib                      1.2.13               h53f4e23_5    conda-forge
zstd                      1.5.5                h4f39d0f_0    conda-forg
~~~

And everything works fine after installing the package from this PR:
~~~
(pygame) icub@iCubs-Mac-mini ~ % python -c "import pygame;pygame.font.init()" 
pygame 2.5.2 (SDL 2.28.5, Python 3.12.2)
Hello from the pygame community. https://www.pygame.org/contribute.html
~~~

So I consider the PR ready for review. @conda-forge/sdl2_ttf

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,9 +7,11 @@ package:
 source:
   url: https://github.com/libsdl-org/SDL_ttf/archive/refs/tags/release-{{ version }}.tar.gz
   sha256: 2275d0ddfffa53f0efa628bc1621f662dacbd42467b5a44db99e38255fbb575a
+  patches:  # [osx]
+    - 349.patch  # [osx]
```
Comment (h-vetinari) [inline recipe/meta.yaml]: Please put the batch in a separate folder and document in the recipe where it comes from
```suggestion
  patches:
    # backport of https://github.com/libsdl-org/SDL_ttf/pull/349
    - patches/349.patch  # [osx]
```

**Review by h-vetinari — APPROVED** [review]
LGTM aside from a small nit! Thanks for taking care of this!

**Comment by traversaro** [comment]
Done, thanks!

**Review by h-vetinari — APPROVED** [review]

**Comment by traversaro** [comment]
Sorry, I did not noticed that you also moved the patch location, my bad.

**Pull request merged**